mantidproject/mantid
# Fix EngDiffUI not converting peak centres from TOF to d post sequential fit

**Description of work.**

Sequential fits did not convert peak centres into dSpacing (from TOF). To fix this I have passed the observer the peak centre parameter names as a field in the same dictionary that holds the fit parameters/function.

I have refactored the code for the conversion with two aims:
(1) Future proof the unit conversion to handle difa and tzero diffractometer constants when converting both the centre and error from TOF to dSpacing.
(2) Make the workspace that holds the peak centres in dSpacing a matrix workspace like the fitted parameters (which simplifies the code to process and pass fit results).

I have also fixed a bug when loading data without a proton_charge log value (and a check that the integrated proton charge > 0)


**To test:**
(1) Load in TOF data from PR #29726 and perform a sequential fit as per the instructions.
(2) Add different types of peak (e.g. Gaussian and Lorentzian) and run the sequential fit.
(3) There should be an additional matrix workspace in the group of fit workspaces (suffix `_fit`) for each peak centre parameter which should have been converted into d Spacing (suffix `_dSpacing`).


*There is no associated issue.*


~~*This does not require release notes* because **sequential fitting was a new feature introduced this release and the bug wasn't reported**~~
This got delayed a release - so notes have been added




---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
